src/internal/connector/onedrive/data_collections.go

```diff
@@ -127,11 +130,20 @@ func DataCollections(
 // adds data migrations to the collection set.
 func migrationCollections(
 	svc graph.Servicer,
+	priorVersion int,
 	tenant string,
 	user common.IDNamer,
 	su support.StatusUpdater,
 	ctrlOpts control.Options,
 ) ([]data.BackupCollection, error) {
+	if priorVersion < 1 {
```

Comment:
this seems a bit arbitrary. Can you add a comment for why it's v1? It seems like persisting metadata was enabled prior to versioning so in theory there's v0 backups that could be selected successfully as a base

Though on that note, I think there's a bug in this code. We should only allow reusing the delta token in the version is high enough to have files using IDs instead of names... going to make a ticket for that...

Reply:
I think your mind is on a different topic.  This isn't for metadata handling, specifically.  Per migrations, I'm using < 1 to interpret `no prior backup, nothing to migrate`.  (will add that as a comment)

Reply:
>This isn't for metadata handling, specifically. Per migrations, I'm using < 1 to interpret no prior backup, nothing to migrate. (will add that as a comment)

if it's tied to the backup version though it's possible to have a backup of v0 or v1 that will be selected as a base for incrementals. In those cases we still need to migrate the subtree. There's other reasons why we shouldn't be using those as a base, but it's still possible to select them